projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/JobUpload.java

```diff
@@ -67,7 +75,6 @@ public JobUpload(
    * @return resource containing data job content in a zip format.
    */
   public Optional<Resource> getDataJob(String jobName) {
-    CredentialsProvider credentialsProvider = gitCredentialsProvider.getProvider();
```

Comment:
The getProvider function generates a simple CredentialProvider based on username and password which we are supplying from helm/properties, so I concluded that it's OK to call it once in constructor, rather than every time -

https://github.com/vmware/versatile-data-kit/blob/8c67726e571a66f4624bb6ccd1bf35ae37c60fae/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/GitCredentialsProvider.java#L30

Same for the AWS code commit credential provider, we just need roleARN at the starting which we supplied using datajobs.aws.roleArn property